torchmetrics/functional/audio/pit.py

```diff
@@ -0,0 +1,181 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from importlib.util import find_spec
+from itertools import permutations
+from typing import Any, Callable, Dict, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+_ps_dict: dict = {}  # cache
+_ps_idx_dict: dict = {}  # cache
+
+_has_scipy = find_spec("scipy")
```

Comment:
it is already here
https://github.com/PyTorchLightning/metrics/blob/591bb2d7f92985efb321b54f96b2628c79e56fe5/torchmetrics/utilities/imports.py#L78

Reply:
see use _SCIPY_AVAILABLE: https://github.com/PyTorchLightning/metrics/pull/384/commits/16d70f9385253fdf89f4a0acb0432105dc0080d6